src/main/java/org/graylog/plugins/threatintel/adapters/spamhaus/SpamhausEDROPDataAdapter.java

```diff
@@ -59,7 +59,10 @@ public SpamhausEDROPDataAdapter(@Assisted("id") String id,
     public void doStart() throws Exception {
         final ImmutableMap.Builder<String, Map<SubnetUtils.SubnetInfo, String>> builder = ImmutableMap.builder();
         for (String list : lists) {
-            builder.put(list, fetchSubnetsFromEDROPLists(list));
+            final Map<SubnetUtils.SubnetInfo, String> subnetMap = fetchSubnetsFromEDROPLists(list);
+            if (subnetMap != null) {
```

Comment:
Doesn't the same problem exist in the `doRefresh()` method (line 74)?

Reply:
Would it solve the problem if `fetchSubnetsFromEDROPLists()` returned an empty map instead of `null` when nothing changed?

Reply:
How do we differentiate between the retrieval of an empty and a non-changing list?

Reply:
The `doRefresh()` does not suffer from the same problem, because it is using a `HashMap` instead of an `ImmutableMap.Builder`. In that part of the code it is iterating over the null entries afterwards to populate them with the last retrieved versions of them.

Reply:
Yes, if we return `null`, we do not update the list and use the version of the list that was retrieved successfully last time. If we would return an empty map, it would replace the list with an empty one. It is an edge case that one of the two lists might be emptied, but it is possible.